### ISSUE TITLE
Specific guidance for Mobile Application usage

### ISSUE THREAD

**Issue by ThisIsMissEm** [oauth-wg/draft-ietf-oauth-client-id-metadata-document#21]
With mobile applications, unless you use application claimed https URLs and the system authentication mechanisms (I believe Apple and Google have a "sign in with" dialog for this now, instead of using an in-app webview), then there is potential for the `redirect_uris` to be hijacked by another application (e.g., if using a `myapp://` redirect URI scheme.

I'm not sure what can be done to further mitigate this, besides DPoP and PKCE which mitigate redirect hijacking attempts.

**Comment by aaronpk** [comment]
Is there anything unique about this draft in regards to mobile apps? The custom scheme redirect URI issue applies to all OAuth clients, whether or not they provide their own metadata.

**Comment by ThisIsMissEm** [comment]
Yeah, I suspect there is just the same security considerations as with any custom scheme redirect URI + mobile apps, but it may not be a bad idea to reinforce that for mobile applications, you definitely want to use PKCE if using this I-D?

Or maybe we just cross link to the OAuth 2.0 Security Best Current Practice document as the first section in Security Considerations, highlighting a few of the possible attack vectors to be aware of?

**Comment by ThisIsMissEm** [comment]
So like in the opening to `6. Security Considerations` explicitly listing some of the considerations to pay attention to.

**Comment by aaronpk** [comment]
We could add a reference to [Attestation-Based Client Authentication](https://datatracker.ietf.org/doc/draft-ietf-oauth-attestation-based-client-auth/) to suggest a way to use a native app platform attestation with this spec. I'm not sure if this would be just general guidance, or would require defining an actual mechanism to make work. It feels like if it requires defining a new mechanism that would be best done as a separate document, which I suspect it will. 

I'm thinking things like the `jwks_uri` in the client metadata would point to a public key that the "Client Attester" controls, then the native app would be able to use the platform attestation to call the Client Attester service to create an attestation JWT signed with a key advertised in the `jwks_uri`. That would connect all the dots, but I think that is too much to just throw out there as an example.
